specs/disciplina/educator/api/spec.org

```diff
@@ -0,0 +1,120 @@
+   _enroll_ to a course. There are two possible ways of enrolling the student
+   into a course:
+   
+   1) *Default courses*. As soon as *Student* becomes _known_ to *Educator*, the
+      *Educator* automatically _enrolls_ *Student* to a particular set of
+      courses. It is assumed that this set of courses is defined outside
+      Disciplina system (e. g. in the integrated CRM). Therefore, as soon as
+      *Student* becomes known to *Educator*, the *Student* becomes automatically
+      enrolled to these courses and can interact with a course content (Section
+      3).
+   2) *Student-chosen courses*. As soon as *Student* becomes _known_ to
+      *Educator*, they can query the *Educator* for the list of _available
+      courses_, and selectively _enroll_ to these courses (using Student API).
+      Later a student should still be able to see both list of courses they're
+      enrolled in and list of courses still available for enrollment.
+      
+   These two ways can be _combined_: there can be a set of obligatory default
+   courses together with additional facultative courses.
+
+* Studying the course
+  A course consists of _assignments_, for which *Student* should provide
+  _submissions_. Submissions later receive _grades_ from Educator. _Grades_
+  (which are also private chain _transaction_) are combined into _blocks_ once
+  in _block issue period_.
+  
+  After a _grade_ is added into a _block_, there is a _proof_ (a path in sized
+  Merkle tree) for this grade, which can be fetched by a _Student_.
+
+** Assignments
+  There are two modes of operation for assigning _assignments_ to students:
+
+  1) *Default course assignments*. As soon as *Student* enrolls into a _course_,
+     *Educator* automatically assigns a particular set of this course's
+     _assignments_ to the *Student*. It is assumed that this set of assignments
+     is defined outside Disciplina system (e. g. in the integrated CRM).
+     Therefore, as soon as *Student* enrolls to a course, they can submit
+     submissions for these default assignments.
+  2) *On-demand assignments*. *Educator* assigns assignments to students as the
+     course goes. In the beginning of the course, there might be no course
+     assignments assigned to a *Student*. A *Student* should poll Student API to
+     check for updates regarding course assignments.
+     
+  These two modes can be _combined_: there can be a set of default course
+  assignments, and particular students also can receive additional ones during
+  the course.
+** Submissions
+   There are two modes of operations of submitting _submissions_ to assignments:
+   
+   1) *Offline submissions*. If a course is meant to be offline (e. g. it's a
+      classic university course with course assignments done on paper) and there
+      are no actual digital _content_ of submissions, then submissions in
+      Disciplina system become effectively fictional. This means that these
```

Comment:
Not exactly. They serve the role of protecting the student from the unhonest educator. Since every assignment must be signed by the student before a grade for the submission can be assigned, student can effectively stop the course at any point. Otherwise, educator can at any moment control grades for the future assignments which were not really made by the student.